lib/marc_to_argot/macros/unc/holdings.rb

```diff
@@ -27,13 +27,14 @@ def holdings(rec, cxt)
           unless cxt.clipboard[:shared_record_set] == 'dws'
             # create HoldingsRecord objects to process
             tmp_holdings = create_temp_holdings_for_processing(rec)
-            # process them to extract the necessary data
-            tmp_holdings.each { |hrec| hrec.process_holdings_data }
-            # write them out to argot
-            argotholdings = []
-            tmp_holdings.each { |hrec| argotholdings << hrec.to_argot }
-            cxt.output_hash['holdings'] = argotholdings if argotholdings.length > 0
-#            puts "AH: #{argotholdings}"
+            if tmp_holdings && tmp_holdings.length > 0
+              # process them to extract the necessary data
+              tmp_holdings.each { |hrec| hrec.process_holdings_data }
+              # write them out to argot
+              argotholdings = []
+              tmp_holdings.each { |hrec| argotholdings << hrec.to_argot }
```

Comment:
You can combine lines 34 and 35 by using `map` which will return an array (instead of `each`) :

`argotholdings = tmp_holdings.map(&:to_argot)`

the `&:to_argot` is equivalent to `{ |hrec| hrec.to_argot }` -- call this method on each object.

Reply:
Changed. I don't know why I have such a mental block around the &: idiom, but it will NOT seem to stick in my brain. Never will if I don't start using it!